code/game/objects/items/power_cells.dm

```diff
@@ -13,21 +13,256 @@
+			span_notice("You begin rigging [src] to detonate with [I]."))
+			if(!do_after(user, delay, NONE, src, BUSY_ICON_BUILD))
+				return
+			rigged = TRUE
+			overlays += spark_overlay
+			user.visible_message(span_notice("[user] finishes manipulating [src] with [I]."),
+			span_notice("You rig [src] to explode on use with [I]."))
+		else
+			if(skill < SKILL_ENGINEER_ENGI)
+				user.visible_message(span_notice("[user] fumbles around figuring out how to manipulate [src]."),
+				span_notice("You fumble around, trying to figure out how to stabilize [src]."))
+				var/fumbling_time = SKILL_TASK_EASY
+				if(!do_after(user, fumbling_time, NONE, src, BUSY_ICON_UNSKILLED))
+					return
+				if(prob((SKILL_ENGINEER_PLASTEEL - skill) * 20))
+					to_chat(user, "<font color='danger'>After several seconds of your clumsy meddling [src] buzzes angrily as if offended. You have a <b>very</b> bad feeling about this.</font>")
+					rigged = TRUE
+					explode() //Oops. Now you fucked up (or succeeded only too well). Immediate detonation.
+			user.visible_message(span_notice("[user] begins manipulating [src] with [I]."),
+			span_notice("You begin stabilizing [src] with [I] so it won't detonate on use."))
+			if(skill > SKILL_ENGINEER_ENGI)
+				delay = max(delay - 10, 0)
+			if(!do_after(user, delay, NONE, src, BUSY_ICON_BUILD))
+				return
+			rigged = FALSE
+			overlays -= spark_overlay
+			user.visible_message(span_notice("[user] finishes manipulating [src] with [I]."),
+			span_notice("You stabilize the [src] with [I]; it will no longer detonate on use."))
+
+/obj/item/cell/emp_act(severity)
+	. = ..()
+	charge = max(charge - ((maxcharge * 0.5) / severity), 0)
+	update_appearance(UPDATE_ICON)
+
+/obj/item/cell/ex_act(severity)
+	switch(severity)
+		if(EXPLODE_DEVASTATE)
+			qdel(src)
+		if(EXPLODE_HEAVY)
+			if(prob(50))
+				qdel(src)
+				return
+			if(prob(50))
+				corrupt()
+		if(EXPLODE_LIGHT)
+			if(prob(25))
+				qdel(src)
+				return
+			if(prob(25))
+				corrupt()
+		if(EXPLODE_WEAK)
+			if(prob(25))
+				corrupt()
+
 /obj/item/cell/suicide_act(mob/user)
 	user.visible_message(span_danger("[user] is licking the electrodes of the [src.name]! It looks like [user.p_theyre()] trying to commit suicide."))
 	return (FIRELOSS)
 
+/obj/item/cell/use(amount) // use power from a cell
+	if(rigged && amount > 0)
+		explode()
+		return FALSE
+	last_use = world.time
+
+	if(charge < amount)
+		return FALSE
+	charge = (charge - amount)
+	return TRUE
+
+///Adds power to the cell
+/obj/item/cell/proc/give(amount)
+	if(rigged && amount > 0)
+		explode()
+		return FALSE
+
+	if(maxcharge < amount)
+		return FALSE
+	var/amount_used = min(maxcharge-charge,amount)
+	charge += amount_used
+	return amount_used
+
+///return % charge of cell
+/obj/item/cell/proc/percent()
+	return 100 * (charge / maxcharge)
+
+///Returns TRUE if charge is equal to maxcharge
+/obj/item/cell/proc/fully_charged()
+	return (charge == maxcharge)
```

Comment:
```suggestion
/obj/item/cell/proc/is_fully_charged()
	return charge == maxcharge
```
I recommend changing this slightly misleading proc name. Also the parentheses don't have any effect here.